File: JoinDemo/app/src/main/java/com/baiheng/joindemo/MainActivity.java
package com.baiheng.joindemo;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FrameLayout frameLayout = (FrameLayout) findViewById(R.id.id_container);
        frameLayout.addView(new CustomView(MainActivity.this));

    }
}

File: DataBindingDemo/app/src/main/java/com/zhewen/databindingdemo/MainActivity.kt
package com.zhewen.databindingdemo

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ObservableArrayList
import androidx.databinding.ObservableArrayMap
import androidx.databinding.ObservableInt
import com.zhewen.databindingdemo.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity(),View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //        setContentView(R.layout.activity_main);
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(
            this,
            R.layout.activity_main
        )
        val user = User("西门吹雪",22)
//        binding.user = user
//        user.name.set("东方不败")
        val map = ObservableArrayMap<String, String>().apply { put("name", "东方不败") }
        binding.map = map
        val list = ObservableArrayList<String>().apply { add("22") }
        binding.list = list
        binding.key = "name"
        binding.index = 0
        val viewModel:DataViewModel = DataViewModel()
        viewModel.nameLiveData.value = "独孤求败"
        binding.viewmodel = viewModel
        binding.lifecycleOwner = this
        binding.click = this
    }

    override fun onClick(v: View?) {
        Toast.makeText(this@MainActivity, "hahah",Toast.LENGTH_LONG).show()
    }
}